docs/usage/10.md

```diff
@@ -14,9 +14,70 @@ The phases where hooks may be run are:
 # Default hook image
 The default hook image is `quay.io/konveyor/hook-runner:latest`. It is based off of the Ansible Runner image with the addition of python-openshift to provided Ansible k8s resources as well as a recent oc binary.
 
+# Writing ansible for hooks
+When writing ansible to run within the pod it is possible to use the `oc` command or the k8s modules to accomplish tasks. A token is mounted in the container for the specified service account so there is no need to manually authenticate before running tasks in the cluster.
+
+```
+- hosts: localhost
+  gather_facts: false
+  tasks:
+  - name: get pod name
+    shell: oc get po --all-namespaces
+```
+
+```
+- hosts: localhost
+  gather_facts: false
+  tasks:
+  - name: Get pod
+    k8s_facts:
+      kind: pods
+      api: v1
+      namespace: openshift-migration
+      name: "{{ lookup( 'env', 'HOSTNAME') }}"
+    register: pods
+
+  - name: Print pod name
+    debug:
+      msg: "{{ pods.resources[0].metadata.name }}"
+```
+
+# Special Environment variables
+The migplan name and associated namespaces are passed in as environment variables to the job container so they can be used in writing tasks. To access use an env lookup.
+
+For example:
+```
+- hosts: localhost
+  gather_facts: false
+  tasks:
+  - debug:
+      msg: "{{ (lookup( 'env', 'MIGRATION_NAMESPACES')).split(',') }}"
+
+  - debug:
+      msg: "{{ lookup( 'env', 'MIGRATION_PLAN_NAME') }}"
+```
+
 # Hook execution
 When an Ansible playbook is provided as part of a migration hook it will be mounted into the hook container as a ConfigMap. In either case the hook container will be run as job on the desired cluster, using the execution ServiceAccount and Namespace specified. Jobs are designed to run to completion, even if the initial pod is evicted or killed, and may attempt to run several times up until it it either reaches the backoffLimit or successful completion. We use the default backoffLimit of 6 for hook jobs.
 
+# Hook Success/Failure
+Hooks are run as a job and success and failure is detected in the same way, based on the container exit status. The `fail` module can produce a non-zero exit status when detecting cases where a non-zero exit status would not normally be produced, to ensure the hook task fails.
```

Comment:
_Suggest_

"Hooks are run as a job and success and failure is detected in the same way, based on the container exit status"
->
"Hooks are run as Jobs and success and failure is detected based on the Job container exit status"

Reply:
Sidenote: I think words like `Pod` and `Job` are typically capitalized in docs, see https://kubernetes.io/docs/concepts/workloads/controllers/job/ for reference.